lms/djangoapps/course_goals/tests/test_api.py

```diff
@@ -55,6 +55,13 @@ def test_add_invalid_goal(self):
         self.assertEqual(response.status_code, 400)
         self.assertEqual(len(CourseGoal.objects.filter(user=self.user, course_key=self.course.id)), 0)
 
+    def test_add_without_goal_key(self):
+        """ Ensures if no goal key provided, post does not succeed. """
+
+        response = self.post_course_goal(goal_key=None)
+        self.assertEqual(response.status_code, 400)
```

Comment:
Can we also verify the response message?

Reply:
Done